flows/field.go

```diff
@@ -23,7 +23,7 @@ func (f *Field) Asset() assets.Field { return f.Field }
 
 // Value represents a value in each of the field types
 type Value struct {
-	Text     types.XText      `json:"text"`
+	Text     types.XText      `json:"text" validate:"required"`
```

Comment:
So this handles the validation case coming in from JSON but if our contract is not this can't be an empty string we should probably add some assertions or errors on new value creation. (and perhaps make these fields private in the process?)

If my understanding is correct I believe these changes would not work correctly as mailroom is, but no error of validation would show that. (IE, we pass in empty `Value` structs per previous implementation)

Reply:
Yeah we can validate these

Reply:
I've added a check in `NewFieldValues` which already can return an error. I can encapsulate `Value` a bit more if you think it's worth it (interface?) but I just tried using the inner anon struct approach and couldn't figure out how to initialize it in `NewValue`.